src/parser/jobs/ast/modules/EarthlyStar.tsx

```diff
@@ -23,46 +22,50 @@ const SEVERETIES = {
 }
 
 // const PLAYER_CASTS = [ACTIONS.EARTHLY_STAR.id, ACTIONS.STELLAR_DETONATION.id]
-const PET_CASTS = [ACTIONS.STELLAR_BURST.id, ACTIONS.STELLAR_EXPLOSION.id]
 // const PLAYER_STATUSES = [STATUSES.EARTHLY_DOMINANCE.id, STATUSES.GIANT_DOMINANCE.id]
 
 export default class EarthlyStar extends Module {
 	static handle = 'earthlystar'
 	static title = t('ast.earthly-star.title')`Earthly Star`
 
+	@dependency private data!: Data
 	@dependency private suggestions!: Suggestions
 
-	_uses = 0
-	_lastUse = 0
-	_totalHeld = 0
-	_earlyBurstCount = 0
+	private uses = 0
+	private lastUse = 0
+	private totalHeld = 0
+	private earlyBurstCount = 0
+
+	private PET_CASTS: number[] = []
+
 
 	protected init() {
-		this.addEventHook('cast', {abilityId: ACTIONS.EARTHLY_STAR.id, by: 'player'}, this._onSet)
-		this.addEventHook('cast', {abilityId: PET_CASTS, by: 'pet'}, this._onPetCast)
-		// this.addEventHook('applybuff', {abilityId: PLAYER_STATUSES, by: 'player'}, this._onDominance)
+		this.PET_CASTS = [this.data.actions.STELLAR_BURST.id, this.data.actions.STELLAR_EXPLOSION.id]
+		this.addEventHook('cast', {abilityId: this.data.actions.EARTHLY_STAR.id, by: 'player'}, this.onSet)
+		this.addEventHook('cast', {abilityId: this.PET_CASTS, by: 'pet'}, this.onPetCast)
+		// this.addHook('applybuff', {abilityId: PLAYER_STATUSES, by: 'player'}, this._onDominance)
 
 		this.addEventHook('complete', this._onComplete)
 	}
 
-	_onSet(event: CastEvent) {
-		this._uses++
+	private onSet(event: CastEvent) {
+		this.uses++
 		// TODO: Instead determine how far back they used it prepull by checking explosion time.
-		if (this._lastUse === 0) { this._lastUse = this.parser.fight.start_time }
+		if (this.lastUse === 0) { this.lastUse = this.parser.fight.start_time }
 
-		const _held = event.timestamp - this._lastUse - (ACTIONS.EARTHLY_STAR.cooldown * 1000)
-		if (_held > 0) {
-			this._totalHeld += _held
+		const held = event.timestamp - this.lastUse - (this.data.actions.EARTHLY_STAR.cooldown * 1000)
+		if (held > 0) {
+			this.totalHeld += held
 		}
```

Comment:
Do you intend to handle delayed Earthly? I don't know if AST usually pre-pulls it so may want to move the `this.uses` set to later in the function or check against `this.lastUse` and handle initial cast relative to fight pull time.

Reply:
Oh yeah, we have prepull support now. I'll make a change